### DISCUSSION TITLE
Commit query notice or button to commit query when auto-commit checkbox disabled

### DISCUSSION THREAD

**Discussion by daevidvincent** [dbeaver/dbeaver#14325] [General]
**Is your feature request related to a problem? Please describe.**
I just wasted literally hours performing some updates on 6 x SDLC environments to `UPDATE` three tables with millions of rows that takes 45 minutes each to complete! This is because if "auto-commit" is OFF, the query LOOKS like it does something, but actually didn't do anything!

![image](https://user-images.githubusercontent.com/20825058/76130923-55bd0800-5fc1-11ea-823a-cdbf7efe931b.png)

![image](https://user-images.githubusercontent.com/20825058/76130879-302ffe80-5fc1-11ea-96c7-fa07e6abc87b.png)

I don't see the purpose of this checkbox when you already have a `read-only` connection to prevent writing.

![image](https://user-images.githubusercontent.com/20825058/76130901-450c9200-5fc1-11ea-9f3e-23a9f7071751.png)

**Describe the solution you'd like**
If AUTO-COMMIT is UNCHECKED, 

1. put some kind of indicator in the bottom status bar that this is a "read only" connection or however you want to word it.
2. after a query that requires `COMMIT` make it obvious with a big red banner or something in the result set output, or at the bottom this has NOT been commited.
3. in the result set where it misleadingly says, "Updated Rows: 814899" it should change to "Potentially Updated Rows: 814899 -- COMMIT changes pending" or something like that
4. give me a button to push to actually perform the `COMMIT` or a `yes`/`no` dialog box or something! Maybe make the "COMMIT changes pending" in #3 a link that does the action? 

**Describe alternatives you've considered**
I've considered punching a baby in the face because I have had to stay up late to run these queries after multi-timezones have gone to sleep and so to be less disruptive...

**Additional context**
Think of the children! Think how many hours of frustration you would save pleebs like me who apparently are too stupid to know that they have to remember that this checkbox is off and now have to `COMMIT` their changes, yet I don't even know how or where to do that when running a query such as this...

![image](https://user-images.githubusercontent.com/20825058/76131146-31156000-5fc2-11ea-8704-6e9f0118ccec.png)

**Comment by uslss** [comment]
thanks for suggestion

**Comment by GeriZ** [comment]
Hi,

you can see the commit status of your connection in SQL editor. When the commit and rollback buttons are greyed out then autocommit is active. When they are not greyed out then autocommit is off.

![image](https://user-images.githubusercontent.com/31589944/78997064-2d877280-7b46-11ea-903e-9252fe8f5ede.png)


Regards
Geri
